src/main/host/descriptor/socket/unix.rs

```diff
@@ -0,0 +1,400 @@
+        self.peer_addr
+    }
+
+    pub fn address_family(&self) -> nix::sys::socket::AddressFamily {
+        nix::sys::socket::AddressFamily::Unix
+    }
+
+    pub fn recv_buffer(&self) -> &Arc<AtomicRefCell<SharedBuf>> {
+        &self.recv_buffer
+    }
+
+    pub fn close(&mut self, event_queue: &mut EventQueue) -> SyscallResult {
+        // drop the event listener handles so that we stop receiving new events
+        self.send_buffer_event_handle
+            .take()
+            .map(|h| h.stop_listening());
+        self.recv_buffer_event_handle
+            .take()
+            .map(|h| h.stop_listening());
+
+        // inform the buffer that there is one fewer writers
+        if let Some(send_buffer) = self.send_buffer.as_ref() {
+            send_buffer.borrow_mut().remove_writer(event_queue);
+        }
+
+        // no need to hold on to the send buffer anymore
+        self.send_buffer = None;
+
+        // set the closed flag and remove the active, readable, and writable flags
+        self.copy_state(
+            FileState::CLOSED | FileState::ACTIVE | FileState::READABLE | FileState::WRITABLE,
+            FileState::CLOSED,
+            event_queue,
+        );
+
+        Ok(0.into())
+    }
+
+    pub fn read<W>(
+        &mut self,
+        mut _bytes: W,
+        _offset: libc::off_t,
+        _event_queue: &mut EventQueue,
+    ) -> SyscallResult
+    where
+        W: std::io::Write + std::io::Seek,
+    {
+        // we could call recvfrom() here, but for now we expect that there are no code paths that
+        // would get here
+        panic!("Called read() on a unix socket");
+    }
+
+    pub fn write<R>(
+        &mut self,
+        mut _bytes: R,
+        _offset: libc::off_t,
+        _event_queue: &mut EventQueue,
+    ) -> SyscallResult
+    where
+        R: std::io::Read + std::io::Seek,
+    {
+        // we could call sendto() here, but for now we expect that there are no code paths that
+        // would get here
+        panic!("Called write() on a unix socket");
+    }
+
+    pub fn sendto<R>(
+        &mut self,
+        mut bytes: R,
+        addr: Option<nix::sys::socket::SockAddr>,
+        event_queue: &mut EventQueue,
+    ) -> SyscallResult
+    where
+        R: std::io::Read + std::io::Seek,
+    {
+        // if the file is not open for writing, return EBADF
+        if !self.mode.contains(FileMode::WRITE) {
+            return Err(nix::errno::Errno::EBADF.into());
+        }
+
+        let send_buffer = match (&self.send_buffer, addr) {
+            // already connected but a destination address was given
+            (Some(send_buffer), Some(_addr)) => match self.socket_type {
+                UnixSocketType::Stream => return Err(Errno::EISCONN.into()),
+                UnixSocketType::SeqPacket => send_buffer,
+                UnixSocketType::Dgram => {
+                    // TODO: need to properly look up the socket once we support named sockets
```

Comment:
Maybe worth logging a warning, since this could be the point where a simulation starts diverging from "real" behavior?

Reply:
Added a warning.